src/module-elasticsuite-catalog-optimizer/Controller/Adminhtml/Optimizer/Duplicate.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Smile ElasticSuite to newer
+ * versions in the future.
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteCatalogOptimizer
+ * @author    Dmytro ANDROSHCHUK <dmand@smile.fr>
+ * @copyright 2020 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+namespace Smile\ElasticsuiteCatalogOptimizer\Controller\Adminhtml\Optimizer;
+
+use Smile\ElasticsuiteCatalogOptimizer\Controller\Adminhtml\AbstractOptimizer as OptimizerController;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Optimizer Adminhtml duplicate controller.
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteCatalogOptimizer
+ * @author   Dmytro ANDROSHCHUK <dmand@smile.fr>
+ */
+class Duplicate extends OptimizerController
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function execute()
+    {
+        /** @var \Magento\Backend\Model\View\Result\Page $resultPage */
+        $resultPage = $this->resultPageFactory->create();
+
+        $optimizerId = (int) $this->getRequest()->getParam('id');
+        $optimizer = null;
+
+        try {
+            $optimizer = $this->optimizerRepository->getById($optimizerId);
+            $this->coreRegistry->register('current_optimizer', $optimizer);
+            $this->coreRegistry->register('is_duplicate', true);
+            $resultPage->getConfig()->getTitle()->prepend(__('Duplicate %1', $optimizer->getName()));
```

Comment:
This controller should not have his own layout and page rendering.

It would be more simple to do like Magento does with products duplication : https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Catalog/Controller/Adminhtml/Product/Duplicate.php#L60